doc/mgmt/Sonic_Config_Mgmt_HLD.md

```diff
@@ -0,0 +1,187 @@
+# Feature Name
+Implement management support using CLI/REST/gNOI interfaces for Configuration Management Operations.
+
+# High Level Design Document
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 10/30/2019  |  Bhavesh Nisar     | Initial version                   |
+
+# About this Manual
+This document provides information on the management interfaces for Configuration Management Operations.
+# Scope
+The scope of this document is limited to the northbound interfaces supported by the new management framework, i.e., CLI, gNOI and REST. The functional behavior of these operations is not modified and can be referred in the SONiC Documents.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+| NBI                      | North Bound Interface               |
+
+# 1 Feature Overview
+
+The feature addresses the equivalent of the existing cLick CLI commands.  The  implementation is contained within the Management Framework container.
+Their is no openconfig data yang defined for this feature. A new SONiC yang will be introduced for the NBI.
+
+## 1.1 Requirements
+
+### 1.1.1 Functional Requirements
+Not Applicable
+
+### 1.1.2 Configuration and Management Requirements
+
+Operations:
+1. Save running configuration to default. <br>
+   CLI Command : write memory <br>
+   EXEC Level <br>
+   Write to file /etc/sonic/config_db.json  <br>
+
+2. Copy configuration from file to running and reload. <br>
```

Comment:
When "config reload" is executed, all the containers are reloaded. Does it mean the sonic-mgmt-framework container is also reloaded. This may kill the "copy" session initiated. Ensure that even if the session is terminated, user gets some feedback and does not loose connectivity forever. Also care needs to be taken that the "config reload" session initiated on the host o/s finishes and is not interrupted because management-container is reastarted.